sql-plugin/src/main/java/com/nvidia/spark/rapids/GpuColumnVector.java

```diff
@@ -107,44 +107,61 @@ public static synchronized void debug(String name, HostColumnVector hostCol) {
               hexString(hostCol.getUTF8(i)));
         }
       }
-    } else if (DType.INT32.equals(type)) {
-      for (int i = 0; i < hostCol.getRowCount(); i++) {
-        if (hostCol.isNull(i)) {
-          System.err.println(i + " NULL");
-        } else {
-          System.err.println(i + " " + hostCol.getInt(i));
-        }
-      }
-    } else if (DType.INT8.equals(type)) {
-      for (int i = 0; i < hostCol.getRowCount(); i++) {
-        if (hostCol.isNull(i)) {
-          System.err.println(i + " NULL");
-        } else {
-          System.err.println(i + " " + hostCol.getByte(i));
-        }
-      }
-    } else if (DType.BOOL8.equals(type)) {
+    } else if (DType.INT32.equals(type)
+            || DType.INT8.equals(type)
+            || DType.INT16.equals(type)
+            || DType.INT64.equals(type)
+            || DType.TIMESTAMP_DAYS.equals(type)
+            || DType.TIMESTAMP_SECONDS.equals(type)
+            || DType.TIMESTAMP_MICROSECONDS.equals(type)
+            || DType.TIMESTAMP_MILLISECONDS.equals(type)
+            || DType.TIMESTAMP_NANOSECONDS.equals(type)) {
+      debugInteger(hostCol, type);
+   } else if (DType.BOOL8.equals(type)) {
       for (int i = 0; i < hostCol.getRowCount(); i++) {
         if (hostCol.isNull(i)) {
           System.err.println(i + " NULL");
         } else {
           System.err.println(i + " " + hostCol.getBoolean(i));
         }
       }
-    } else if (DType.TIMESTAMP_MICROSECONDS.equals(type) ||
-        DType.INT64.equals(type)) {
-      for (int i = 0; i < hostCol.getRowCount(); i++) {
-        if (hostCol.isNull(i)) {
-          System.err.println(i + " NULL");
-        } else {
-          System.err.println(i + " " + hostCol.getLong(i));
-        }
-      }
     } else {
       System.err.println("TYPE " + type + " NOT SUPPORTED FOR DEBUG PRINT");
     }
   }
 
+  private static void debugInteger(HostColumnVector hostCol, DType intType) {
+    for (int i = 0; i < hostCol.getRowCount(); i++) {
+      if (hostCol.isNull(i)) {
+        System.err.println(i + " NULL");
+      } else {
+        String value;
+        switch (intType.getTypeId()) {
+          case INT8:
```

Comment:
Given this is only supposed to be called when the caller already knows the type is an integer, I think we can avoid redundantly switching on every possible enum and instead leverage `DType#getSizeInBytes` to know whether we need to get a byte, short, int, or long.